src/data/finset/preimage.lean

```diff
@@ -82,6 +82,27 @@ lemma image_preimage_of_bij [decidable_eq β] (f : α → β) (s : finset β)
   image f (preimage s f hf.inj_on) = s :=
 finset.coe_inj.1 $ by simpa using hf.image_eq
 
+lemma preimage_subset {f : α ↪ β} {s : finset β} {t : finset α} {h : set.inj_on f (f ⁻¹' s)}
```

Comment:
`h` is provable using just the type of `f`, right? And the statement will be just as general by proving it inline because of proof irrelevance